src/configs/transpose-config.c

```diff
@@ -214,10 +214,41 @@ static void init_transpose_config(void) {
       .const_size_ukernel = (xnn_transposec_ukernel_fn) xnn_x24_transposec_ukernel__1x2_scalar,
       .tile_size = 32,
     };
-    transpose_config.x32 = (struct xnn_transpose_subconfig) {
-      .const_size_ukernel = (xnn_transposec_ukernel_fn) xnn_x32_transposec_ukernel__2x4_scalar_int,
-      .tile_size = 32,
-    };
+    #if XNN_ENABLE_RISCV_VECTOR
+      const struct xnn_hardware_config* hardware_config = xnn_init_hardware_config();
+      assert(hardware_config != NULL);
+      if (hardware_config->vlenb >= 128) {
+        transpose_config.x32 = (struct xnn_transpose_subconfig) {
+          .const_size_ukernel = (xnn_transposec_ukernel_fn) xnn_x32_transposec_ukernel__32x8_rvv,
+          .tile_size = 32,
+        };
+      } else if (hardware_config->vlenb == 64) {
+        transpose_config.x32 = (struct xnn_transpose_subconfig) {
+          .const_size_ukernel = (xnn_transposec_ukernel_fn) xnn_x32_transposec_ukernel__16x8_rvv,
+          .tile_size = 32,
+        };
+      } else if (hardware_config->vlenb == 32) {
+        transpose_config.x32 = (struct xnn_transpose_subconfig) {
+          .const_size_ukernel = (xnn_transposec_ukernel_fn) xnn_x32_transposec_ukernel__8x8_rvv,
+          .tile_size = 32,
+        };
+      } else if (hardware_config->vlenb == 16) {
+        transpose_config.x32 = (struct xnn_transpose_subconfig) {
+          .const_size_ukernel = (xnn_transposec_ukernel_fn) xnn_x32_transposec_ukernel__4x4_rvv,
+          .tile_size = 32,
+        };
+      } else {
+        transpose_config.x32 = (struct xnn_transpose_subconfig) {
+          .const_size_ukernel = (xnn_transposec_ukernel_fn) xnn_x32_transposec_ukernel__2x4_scalar_int,
```

Comment:
Is scalar needed?  Can't we use 4x4_rvv for all other vector sizes?

Reply:
When vlenb is 8, according to the rule, the maximum tile size would be 2x2. However, this tile size is too small, falling back to scalar would be preferable.